src/main/java/com/woowacamp/soolsool/core/cart/domain/Cart.java

```diff
@@ -1,11 +1,68 @@
 package com.woowacamp.soolsool.core.cart.domain;
 
-import java.util.Set;
-import lombok.RequiredArgsConstructor;
+import com.woowacamp.soolsool.core.cart.code.CartItemErrorCode;
+import com.woowacamp.soolsool.global.exception.SoolSoolException;
+import java.util.List;
+import java.util.Objects;
+import lombok.Getter;
 
-@RequiredArgsConstructor // TODO: 컴파일 에러 방지용 임시 생성자
+@Getter
 public class Cart {
 
+    private static final int MAX_CART_SIZE = 100;
+
     private final Long memberId;
-    private final Set<CartItem> cartItems;
+    private final List<CartItem> cartItems;
+
+    public Cart(final Long memberId, final List<CartItem> cartItems) {
+        validateMember(memberId, cartItems);
+
+        this.memberId = memberId;
+        this.cartItems = cartItems;
+    }
+
+    private void validateMember(final Long memberId, final List<CartItem> cartItems) {
+        boolean hasNotSameMember = cartItems.stream()
+            .anyMatch(cartItem -> !Objects.equals(memberId, cartItem.getMemberId()));
+
+        if (hasNotSameMember) {
+            throw new SoolSoolException(CartItemErrorCode.NOT_EQUALS_MEMBER);
+        }
+    }
+
+    public void addCartItem(final CartItem newCartItem) {
+        validateNull(newCartItem);
+        validateExceedMaxSize();
+        validateDuplicated(newCartItem);
+        validateLiquorStatus(newCartItem);
+
+        cartItems.add(newCartItem);
+    }
+
+    private void validateNull(final CartItem newCartItem) {
+        if (newCartItem == null) {
+            throw new SoolSoolException(CartItemErrorCode.NULL_LIQUOR);
+        }
+    }
+
+    private void validateExceedMaxSize() {
+        if (cartItems.size() == MAX_CART_SIZE) {
+            throw new SoolSoolException(CartItemErrorCode.EXCEED_MAX_CART_SIZE);
+        }
+    }
+
+    private void validateDuplicated(final CartItem newCartItem) {
+        final boolean hasDuplicatedCartItem = cartItems.stream()
+            .anyMatch(newCartItem::isSameWith);
+
+        if (hasDuplicatedCartItem) {
+            throw new SoolSoolException(CartItemErrorCode.EXISTS_CART_ITEM);
+        }
+    }
+
+    private void validateLiquorStatus(final CartItem newCartItem) {
+        if (newCartItem.getLiquor().isStopped()) {
+            throw new SoolSoolException(CartItemErrorCode.STOPPED_LIQUOR);
+        }
+    }
```

Comment:
```suggestion
    private void validateLiquorStatus(final CartItem newCartItem) {
        if (newCartItem.isStopped()) {
            throw new SoolSoolException(CartItemErrorCode.STOPPED_LIQUOR);
        }
    }
```

`Cart`에서 `Liquor` 정보를 아는 건 [디미터 법칙](https://github.com/whatasame/TIL/blob/main/stub/software-development-methodology/object-oriented-programming/%EB%94%94%EB%AF%B8%ED%84%B0%EC%9D%98%20%EB%B2%95%EC%B9%99(Law%20of%20Demeter).md) 위반 같은데 어떻게 생각하시나요?

Reply:
와방 굳입니다. hasStoppedLiquor() 메서드를 만들어 사용했읍니다!